archaius2-api/src/main/java/com/netflix/config/api/Configuration.java

```diff
@@ -0,0 +1,63 @@
+package com.netflix.config.api;
+
+import java.lang.reflect.Type;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.util.Collection;
+import java.util.Optional;
+
+/**
+ * Top level configuration API for users to read properties. 
+ *
+ * @param <PS>
+ */
+public interface Configuration {
+
+    public Optional<Long> getLong(String key);
+
+    public Optional<String> getString(String key);
+
+    public Optional<Double> getDouble(String key);
+
+    public Optional<Integer> getInteger(String key);
+
+    public Optional<Boolean> getBoolean(String key);
+
+    public Optional<Short> getShort(String key);
+
+    public Optional<BigInteger> getBigInteger(String key);
+
+    public Optional<BigDecimal> getBigDecimal(String key);
+
+    public Optional<Float> getFloat(String key);
+
+    public Optional<Byte> getByte(String key);
+
+    /**
+     * Return a 'raw' property that is neither interpolated nor resolved to a specific type
+     * 
+     * @param key
+     * @return
+     */
+    public Optional<?> getProperty(String key);
+    
+    /**
+     * Get a property that has been interpolated and resolved to a specific type
+     * @param key
+     * @param type
+     * @return
+     */
+    public Optional<Object> get(String key, Type type);
```

Comment:
Return type probably shouldn't be object. Type should probably be ParameterizedType